oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java

```diff
@@ -257,5 +259,26 @@ static PrivateKey privateKeyFromPkcs8(String privateKeyPkcs8) throws IOException
     throw new IOException("Unexpected exception reading PKCS#8 data", unexpectedException);
   }
 
+  /**
+   * Generates a Basic Authentication header string for the provided username and password.
+   *
+   * <p>This method constructs a Basic Authentication string using the provided username and
+   * password. The credentials are encoded in Base64 format and prefixed with the "Basic " scheme
+   * identifier.
+   *
+   * @param username The username for authentication.
+   * @param password The password for authentication.
+   * @return The Basic Authentication header value.
+   * @throws IllegalArgumentException if either username or password is null or empty.
+   */
+  static String generateBasicAuthHeader(String username, String password) {
+    if (username == null || username.isEmpty() || password == null || password.isEmpty()) {
```

Comment:
nit: Perhaps we can use Guava's [isNullOrEmpty](https://guava.dev/releases/32.1.1-jre/api/docs/com/google/common/base/Strings.html#isNullOrEmpty(java.lang.String)) for username and password